### ISSUE TITLE
flake8-print auto fix removes valid code

### ISSUE THREAD

**Issue by serjflint** [charliermarsh/ruff#1269]
```python
with open(output_filepath, "w") as out:
        for row in result:
            print(json.dumps(row), file=out)
```

becomes

```python
with open(output_filepath, "w") as out:
        for _row in result:
            pass
```

**Comment by charliermarsh** [comment]
Is the idea here that we should ignore when writing with a file keyword?

**Comment by charliermarsh** [comment]
(I wonder if those should even be flagged as errors?)

**Comment by squiddy** [comment]
> (I wonder if those should even be flagged as errors?)

`flake8-print` doesn't really give away the intention with respect to the file keyword, but I feel like it is more towards removing superfluous debug statements, rather than completely denying any use of print/pprint.

So I would not even flag these cases as errors.

**Comment by charliermarsh** [comment]
I agree.

**Comment by charliermarsh** [comment]
I can fix this, unless @squiddy has started on it.

**Comment by squiddy** [comment]
Go ahead @charliermarsh :)
